rotkehlchen/chain/evm/decoding/thegraph/balances.py

```diff
@@ -46,15 +43,19 @@ def __init__(
         self.token = native_asset.resolve_to_evm_token()
         self.staking_contract = staking_contract
 
-    def _query_vesting_balances(self, balances: 'BalancesSheetType') -> 'BalancesSheetType':
-        """Query balances of vested GRT tokens if DelegationTransfer events are found.
-        This function modifies and returns the 'balances'
+    def query_balances(self) -> 'BalancesSheetType':
+        """
+        Query balances of GRT tokens delegated to indexers if deposit events are found.
+        First, the current shares amounts are fetched from the contract,
+        then shares are converted into GRT balances according to the current pool ratio.
+        The results include delegation rewards earned over time.
         """
+        balances: BalancesSheetType = defaultdict(BalanceSheet)
+
+        # fetch deposit events
         db_filter = EvmEventFilterQuery.make(
             counterparties=[self.counterparty],
-            location=Location.ETHEREUM,
-            event_types=[HistoryEventType.INFORMATIONAL],
-            event_subtypes=[HistoryEventSubType.NONE],
+            products=[EvmProduct.STAKING],
```

Comment:
why removing ethereum from here? In fact shouldn't you have a logic for ethereum and a logic for arbitrum? Probably sharing most of the code but with some differences

Reply:
They are pretty much the same, I'll add back the location and make it dynamic.